src/core/Slider/Slider.tsx

```diff
@@ -12,6 +12,47 @@ import { getBoundedValue } from '../utils/common';
 import { Track } from './Track';
 import { Thumb } from './Thumb';
 
+// based on https://usehooks.com/useEventListener/
+function usePointerEventListener(
+  eventName: string,
+  handler: (event: PointerEvent) => void,
+  element: HTMLElement | Document | undefined,
+) {
+  // Create a ref that stores handler
+  const savedHandler = React.useRef<(event: PointerEvent) => void>();
+
+  // Update ref.current value if handler changes.
+  // This allows our effect below to always get latest handler ...
+  // ... without us needing to pass it in effect deps array ...
+  // ... and potentially cause effect to re-run every render.
+  React.useEffect(() => {
+    savedHandler.current = handler;
+  }, [handler]);
+
+  React.useEffect(
+    () => {
+      // Make sure element supports addEventListener
+      const isSupported = element && element.addEventListener;
+      if (!isSupported) {
+        return;
+      }
+
+      // Create event listener that calls handler function stored in ref
+      const eventListener = (event: PointerEvent) =>
+        savedHandler.current?.(event);
+
+      // Add event listener
+      element?.addEventListener(eventName, eventListener);
+
+      // Remove event listener on cleanup
+      return () => {
+        element?.removeEventListener(eventName, eventListener);
+      };
```

Comment:
This could be simplified:
```suggestion
      element?.addEventListener(eventName, savedHandler.current);
      return () => element?.removeEventListener(eventName, savedHandler.current);
```

Or does that not work?

Reply:
Okay I think see the reasoning now (from the original source): https://overreacted.io/making-setinterval-declarative-with-react-hooks/#refs-to-the-rescue

I would still remove those redundant comments at least.

Reply:
comments removed